Framework/WorkflowAlgorithms/src/AlignAndFocusPowder.cpp

```diff
@@ -1219,33 +1219,42 @@ void AlignAndFocusPowder::compressEventsOutputWS(const double compressEventsTole
  */
 bool AlignAndFocusPowder::shouldCompressUnfocused(const double compressTolerance, const double tofmin,
                                                   const double tofmax, const bool hasWallClockTolerance) {
-  // compressing isn't an option
-  if (compressTolerance <= 0)
-    return false;
   // compressing to WEIGHTED (w/ time) is harder to predict
   if (hasWallClockTolerance)
     return false;
-
-  // estimate the time-of-flight range for the data
-  // if the parameters aren't supplied, guess 20,000us range
-  const double tofRange = (isEmpty(tofmin) || isEmpty(tofmax)) ? 200000. : std::fabs(tofmax - tofmin);
-
-  // assume one frame although this is generically wrong
-  // there are 3 fields in weighted events no time
-  const double sizeWeightedEventsEstimate = 3. * tofRange / compressTolerance;
+  // compressing isn't an option
+  if (compressTolerance <= 0)
+    return false;
 
   if (const auto eventWS = std::dynamic_pointer_cast<const EventWorkspace>(m_outputW)) {
+    // estimate the time-of-flight range for the data
+    // if the parameters aren't supplied, get them from the workspace
+    double tofmin_wksp = tofmin;
+    double tofmax_wksp = tofmax;
+    if (isEmpty(tofmin) || isEmpty(tofmax))
+      getTofRange(m_outputW, tofmin_wksp, tofmax_wksp);
+    const double tofRange = std::fabs(tofmax_wksp - tofmin_wksp);
+
+    // constants estimating size difference of various events
+    constexpr double TOF_EVENT_FIELDS{2.};
```

Comment:
It struck me that these expressions could probably be defined _automatically_ by their respective class[es].